styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -246,20 +267,24 @@ void examineRunningWFISandAssociatedPods(PodList podList) {
   @VisibleForTesting
   void pollPods() {
     try {
-      final PodList list = client.pods().list();
-      examineRunningWFISandAssociatedPods(list);
-
-      final int resourceVersion = Integer.parseInt(list.getMetadata().getResourceVersion());
-
-      for (Pod pod : list.getItems()) {
-        logEvent(Watcher.Action.MODIFIED, pod, resourceVersion, true);
-        inspectPod(Watcher.Action.MODIFIED, pod);
-      }
+      tryPollPods();
     } catch (Throwable t) {
       LOG.warn("Error while polling pods", t);
     }
   }
 
+  private synchronized void tryPollPods() {
```

Comment:
I guess the reason for the laziness has to do with the ability to reconfigure the target GKE cluster at runtime.